docs/source/ml.rst

```diff
@@ -0,0 +1,182 @@
+****************
+Machine Learning
+****************
+
+CivisML uses the Civis Platform to train machine learning models
+and parallelize their predictions over large datasets.
+It contains best-practice models for general-purpose classification
+and regression modeling as well as model quality evaluations and
+visualizations. All CivisML models use `scikit-learn <http://scikit-learn.org/>`_
+for interoperability with other platforms and to allow you to leverage
+resources in the open-source software community when creating machine learning models.
+
+
+Define Your Model
+=================
+
+Start the modeling process by defining your model. Do this by creating an instance
+of the :class:`~civis.ml.ModelPipeline` class. Each :class:`~civis.ml.ModelPipeline` corresponds to a
+scikit-learn :class:`~sklearn.pipeline.Pipeline` which will run in Civis Platform.
+A :class:`~sklearn.pipeline.Pipeline` allows you to combine multiple
+modeling steps (such as missing value imputation and feature selection) into a
+single model. The :class:`~sklearn.pipeline.Pipeline` is treated as a unit -- for example,
+cross-validation happens over all steps together.
+ 
+You can define your model in two ways, either by selecting a pre-defined algorithm
+or by providing your own scikit-learn
+:class:`~sklearn.pipeline.Pipeline` or :class:`~sklearn.base.BaseEstimator` object.
+Note that whichever option you chose, CivisML will pre-process your data to
+one-hot-encode categorical features (the non-numerical columns) to binary indicator columns
+before sending the features to the :class:`~sklearn.pipeline.Pipeline`.
+
+       
+Pre-Defined Models
+------------------
+
+You can use the following pre-defined models with CivisML.
+All models start by imputing missing values with the mean of non-null
+values in a column. The "sparse_*" models include a LASSO regression step
+(using the `glmnet <https://github.com/civisanalytics/python-glmnet>`_ package)
+to do feature selection before passing data to the final model.
```

Comment:
> The final stage estimator isn't the full story for any of them, since we're also doing missing value imputation

That's a good point.  I think the documentation is clear, we can't avoid that there is a little complexity going on behind the scenes, but it's make everything easier for users.